src/test/java/ioanandroflorea/tests/Tema8/pageObjects/DisabledDynamicButtonsPage.java

```diff
@@ -0,0 +1,67 @@
+package ioanandroflorea.tests.Tema8.pageObjects;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import java.time.Duration;
+
+public class DisabledDynamicButtonsPage {
+
+    ChromeDriver driver;
+    WebDriverWait wait;
+
+    public DisabledDynamicButtonsPage(ChromeDriver driver) {
+        this.driver = driver;
+        wait = new WebDriverWait(driver, Duration.ofSeconds(5));
+        PageFactory.initElements(driver, this);
+    }
+
+    @FindBy(id = "button00")
+    WebElement startButton;
+
+    @FindBy(id = "button01")
+    WebElement oneButton;
+
+    @FindBy(id = "button02")
+    WebElement secondButton;
+
+    @FindBy(id = "button03")
+    WebElement thirdButton;
+
+    public void waitForTheStartButtonToBeClickable() {
+        wait.until(ExpectedConditions.visibilityOf(startButton));
+    }
+
+    public void clickOnStartButton() {
+        startButton.click();
+    }
+
+    public void waitForOneButtonToBeClickable() {
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("button01")));
```

Comment:
Replace visibilityOfElementLocated with elementToBeClickable. The problem is with the expected condition you are using. The elements are already loaded in the page from the beginning so the are visible from the start. What you need here is to wait for them to be clickable as they are disabled until previous buttons are clicked.